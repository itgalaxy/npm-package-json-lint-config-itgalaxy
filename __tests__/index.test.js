const { NpmPackageJsonLint } = require("npm-package-json-lint");
const isPlainObj = require("is-plain-obj");
const config = require("./..");

function lint(packageJsonData, packageConfig) {
  return new NpmPackageJsonLint().lint(packageJsonData, packageConfig);
}

describe("npm-package-json-lint-config-itgalaxy", () => {
  it("npm-package-json-lint config should be an object", () => {
    expect(isPlainObj(config)).toBe(true);
  });

  it("rules should be an object", () => {
    expect(isPlainObj(config.rules)).toBe(true);
  });

  it("npm-package-json-lint should run without failing", () => {
    const packageJsonData = {
      author: "Caitlin Snow"
    };
    const results = lint(packageJsonData, config.rules);

    expect(results.issues).toMatchSnapshot();
  });
});
