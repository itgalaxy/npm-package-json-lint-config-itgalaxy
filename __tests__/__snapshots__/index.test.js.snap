// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm-package-json-lint-config-itgalaxy npm-package-json-lint should run without failing 1`] = `
Array [
  LintIssue {
    "lintId": "require-bugs",
    "lintMessage": "bugs is required",
    "node": "bugs",
    "severity": "error",
  },
  LintIssue {
    "lintId": "require-contributors",
    "lintMessage": "contributors is required",
    "node": "contributors",
    "severity": "error",
  },
  LintIssue {
    "lintId": "require-description",
    "lintMessage": "description is required",
    "node": "description",
    "severity": "error",
  },
  LintIssue {
    "lintId": "require-engines",
    "lintMessage": "engines is required",
    "node": "engines",
    "severity": "error",
  },
  LintIssue {
    "lintId": "require-files",
    "lintMessage": "files is required",
    "node": "files",
    "severity": "error",
  },
  LintIssue {
    "lintId": "require-homepage",
    "lintMessage": "homepage is required",
    "node": "homepage",
    "severity": "error",
  },
  LintIssue {
    "lintId": "require-keywords",
    "lintMessage": "keywords is required",
    "node": "keywords",
    "severity": "error",
  },
  LintIssue {
    "lintId": "require-license",
    "lintMessage": "license is required",
    "node": "license",
    "severity": "error",
  },
  LintIssue {
    "lintId": "require-repository",
    "lintMessage": "repository is required",
    "node": "repository",
    "severity": "error",
  },
]
`;
